contrib/terraform/README.md

```diff
@@ -0,0 +1,79 @@
+# MetalK8s deployment Terraform tooling
+
+This set of [Terraform][1] modules is provided as convenience tooling to deploy
+a fully functional MetalK8s cluster. It currently only supports OpenStack,
+and requires the [OpenStack provider][2] to be configured locally. Feel free
+to submit pull requests if you want to add support for other cloud providers!
+
+[1]: https://www.terraform.io/
+[2]: https://www.terraform.io/docs/providers/openstack/index.html
+
+## How to run
+
+After having configured your cloud provider, you can now configure the
+execution of `terraform apply` using variables. These variables are defined
+in `variables.tf`, along with their description and default values (also see
+[the section below](#Inputs)).
+
+You can pass variables using the command-line. For example, to just deploy a
+Bastion and Bootstrap nodes (without installing MetalK8s), using the default
+network only:
+
+```
+terraform apply -var nodes_count=0 -var private_networks=[]
+```
+
+To manage your deployment configuration more easily, defining these
+variables in a `.tfvars` or `.tfvars.json` file can be useful. More info on
+[Terraform documentation](https://www.terraform.io/docs/configuration/variables.html#variable-definitions-tfvars-files).
+Using such files can be done with `terraform apply -var-file=my.tfvars`.
+
+To deploy a 3 nodes cluster (_only the Bootstrap node being installed with the
+current implementation_), here would be the `.tfvars` to define:
+
+```
+nodes_count = 2
+
+private_networks = ["control_plane", "workload_plane"]
+
+metalk8s_iso_source = "/path/to/iso"
+metalk8s_bootstrap = true
+```
+
+After a `terraform apply`, or `terraform output`, you should see a list of
+IPs for you to connect to the spawned VMs using SSH (with user `centos`).
+An SSH config file will also be generated for convenience, so you can connect
+to nodes using their names:
+
+```
+ssh -F ./ssh_config bootstrap
+```
+
+## Inputs
+
+| Name | Description | Type | Default |
+|------|-------------|:----:|:-----:|
+| `bastion_enabled` | Whether to spawn a Bastion | bool | `"true"` |
+| `bastion_proxy_port` | Port to use for the Bastion HTTP forward proxy \(squid\) | number | `"3128"` |
+| `bastion_setup_tests` | Whether to setup test dependencies on the Bastion | bool | `"false"` |
+| `bastion_test_branch` | Which branch to clone for running tests on the Bastion | string | `""` |
+| `control_plane_subnet` | Existing subnet to use for the control-plane | string | `""` |
+| `control_plane_vip` | VIP to create for the control-plane | string | `"192.168.1.2"` |
```

Comment:
The control-plane doesn't have a VIP (I mean, a network doesn't 'have' some IP address). This should be reworded a bit.